packages/web-app/src/lib/documents-api-wrapper.js

```diff
@@ -0,0 +1,61 @@
+const fetch = require('node-fetch');
+const uuid = require('uuid');
+const { utils } = require('@pins/common');
+
+const config = require('../config/config');
+const parentLogger = require('./logger');
+
+async function handler(path, method = 'GET', opts = {}, headers = {}) {
+  const correlationId = uuid.v4();
+  const url = `${config.documents.url}${path}`;
+
+  const logger = parentLogger.child({
+    correlationId,
+    service: 'Documents Service API',
+  });
+
+  try {
+    logger.debug({ url, method, opts, headers }, 'New call');
+
+    return await utils.promiseTimeout(
+      config.documents.timeout,
+      Promise.resolve().then(async () => {
+        const apiResponse = await fetch(url, {
+          method,
+          headers: {
+            'X-Correlation-ID': correlationId,
+            ...headers,
+          },
+          ...opts,
+        });
+
+        if (!apiResponse.ok) {
+          logger.debug(apiResponse, 'API Response not OK');
```

Comment:
I think it would make the logging more useful if the log message included the current action so that it's clearer where the log message originates from in the code.

For example `Failed to fetch document from Blob Storage`

Reply:
fixed